### PR TITLE
Development

### PR DESCRIPTION
**Pull request by geraldgsh**
- Completed Enumerable module with methods
- Ready for TSE review

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -3,17 +3,179 @@
...
+  # TEST FOR my_select
+  # num = [1, 2, 3, 4, 5, 6, 7, 8]
+  # p num.my_select { |x| x.even? }
+  # p num
+
+  # Task no.4
+  def my_all?
+    if block_given?
+      my_each do |i|
+        return false if yield(self[i]) == false || yield(self[i]).nil?
```
Comment (samgaco) [inline main.rb]: - [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
- [ ] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

File: main.rb [hunk 2]
```diff
@@ -3,17 +3,179 @@
...
+
+  # TEST FOR my_all?
+  # num = [1, 2, 3, 4, 5]
+  # p num.my_all? { |x| x > 6}
+
+  # Task no.5
+  def my_any?
+    if block_given?
+      my_each do |f|
+        return true if yield(f)
```
Comment (samgaco) [inline main.rb]: - [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

File: main.rb [hunk 3]
```diff
@@ -3,17 +3,179 @@
...
+
+  # TEST FOR #my_any?
+  # num = [1, 2, 3, 4, 5, 6]
+  # p num.my_any? { |x| x == 2 }
+
+  # Task no.6
+  def my_none?
+    if block_given?
+      i = 0
+      while self[i]
```
Comment (samgaco) [inline main.rb]: - [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
- [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

File: main.rb [hunk 4]
```diff
@@ -3,17 +3,179 @@
...
+  # TEST my_none
+  # numbers = [1, 2, 3, 4, 5]
+  # p numbers.my_none? {|x| x.is_a?(String) }
+  # p numbers.my_none? {|x| x == 1 }
+
+  # Task no.7
+  def my_count
+    count = 0
+    my_each do |i|
+      count += 1
```
Comment (samgaco) [inline main.rb]: - [ ] my_count counts the number of items in enum that are equal to item if an argument is given
- [ ] my_count counts the number of elements yielding a true value if a block is given

File: main.rb [hunk 5]
```diff
@@ -3,17 +3,179 @@
...
+  end
+
+  # TEST my_count
+  # p ["one", "two", "three", "four", "five"].my_count
+
+  # Task no.8 (This iteration takes a block)
+  def my_map_first
+    if block_given?
+      arr = []
+      my_each do |i|
```
Comment (samgaco) [inline main.rb]: - [ ] my_map returns a new array with the results of running block once for every element in enum.
- [ ] my_map returns an Enumerator if no block is given

File: main.rb [hunk 6]
```diff
@@ -3,17 +3,179 @@
...
+
+  # TEST my_map_first
+  # p ["11", "21", "5", "23", "19"].my_map_first { |str| str.to_i }
+
+  # Task no.9
+
+  def my_inject(_initial = nil)
+    mem = initial
+    my_each do |e|
+      mem = if mem.nil?
```
Comment (samgaco) [inline main.rb]: - [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
- [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @geraldgsh :high_brightness: 

I leave some comments on what functionalities your methods are lacking. You can find a reference of how they are expected to work [here](https://ruby-doc.org/core-2.6.4/Enumerable.html).

Please, work on these required comments and submit them back for another code review once you are done.

------------------------

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Comment by geraldgsh** [comment]
@samgaco noted. Changes have been made and a separate code review request has been made.

**Review by whiz25 — APPROVED** [review]
## Awesome :clap: 
You have implemented all the suggested features. You can merge.

#### Martin Afani

**Comment by geraldgsh** [comment]
> ## Awesome 👏
> You have implemented all the suggested features. You can merge.
> 
> #### Martin Afani

Thanks @whiz25 for taking the time to go thru my work. Have a blessed day ahead of you.

**Pull request merged**